contracts/strategies/StrategyTUSDypool.sol

```diff
@@ -0,0 +1,365 @@
+        uint256 _before = IERC20(ycrv).balanceOf(address(this));
+        yvERC20(yycrv).withdraw(_amt);
+        uint256 _after = IERC20(ycrv).balanceOf(address(this));
+        _amt = _after.sub(_before);
+
+        _before = IERC20(want).balanceOf(address(this));
+        _withdrawOne(_amt);
+        _after = IERC20(want).balanceOf(address(this));
+        tank = tank.add(_after.sub(_before));
+    }
+
+    function drip() public isAuthorized {
+        uint256 _p = yvERC20(yycrv).getPricePerFullShare();
+        _p = _p.mul(ICurveFi(ypool).get_virtual_price()).div(1e18);
+        require(_p >= p, "backward");
+        uint256 _r = (_p.sub(p)).mul(balanceOfYYCRV()).div(1e18);
+        uint256 _s = _r.mul(strategistReward).div(DENOMINATOR);
+        IERC20(yycrv).safeTransfer(strategist, _s.mul(1e18).div(_p));
+        uint256 _t = _r.mul(treasuryFee).div(DENOMINATOR);
+        IERC20(yycrv).safeTransfer(IController(controller).rewards(), _t.mul(1e18).div(_p));
+        p = _p;
+    }
+
+    function tick() public view returns (uint256 _t, uint256 _c) {
+        _t = ICurveFi(ypool).balances(3).mul(yvERC20(ytusd).getPricePerFullShare()).div(1e18).mul(threshold).div(DENOMINATOR);
+        _c = balanceOfYYCRVinWant();
+    }
+
+    function rebalance() public isAuthorized {
+        drip();
+        (uint256 _t, uint256 _c) = tick();
+        if (_c > _t) {
+            _withdrawSome(_c.sub(_t));
+            tank = IERC20(want).balanceOf(address(this));
+            emit Threshold(address(this));
+        }
+    }
+
+    function setGovernance(address _governance) external {
+        require(msg.sender == governance, "!governance");
+        governance = _governance;
+    }
+
+    function setController(address _controller) external {
+        require(msg.sender == governance, "!governance");
+        controller = _controller;
+    }
+
+    function setStrategist(address _strategist) external {
+        require(msg.sender == governance || msg.sender == strategist, "!gs");
+        strategist = _strategist;
+    }
+
+    function setKeeper(address _keeper) external {
+        require(msg.sender == governance || msg.sender == strategist, "!gs");
+        keeper = _keeper;
+    }
+
+    function setWithdrawalFee(uint256 _withdrawalFee) external {
+        require(msg.sender == governance, "!governance");
+        withdrawalFee = _withdrawalFee;
+    }
+
+    function setTreasuryFee(uint256 _treasuryFee) external {
+        require(msg.sender == governance, "!governance");
+        treasuryFee = _treasuryFee;
+    }
+
+    function setStrategistReward(uint256 _strategistReward) external {
+        require(msg.sender == governance, "!governance");
+        strategistReward = _strategistReward;
+    }
+
+    function setThreshold(uint256 _threshold) external {
+        require(msg.sender == keeper || msg.sender == strategist || msg.sender == governance, "!ksg");
+        threshold = _threshold;
+    }
+
+    function setSlip(uint256 _slip) external {
```

Comment:
Same as above

Reply:
Add an additional note. Could also revoke the keeper but the trade-off is to give up resilience to react to any emergency. Always the spectrum of trade-off between every protocol.

Reply:
as discussed with @lbertenasco and @milkyklim i removed the keeper from the setters in this commit a005aec